packages/keycloak-user-management/src/forms/index.tsx

```diff
@@ -0,0 +1,295 @@
+import { KeycloakUser } from '@opensrp/store';
+import { KeycloakService } from '@opensrp/keycloak-service';
+import { Dictionary } from '@onaio/utils/dist/types/types';
+
+/** props for editing a user view */
+export interface UserFormProps {
+  accessToken: string;
+  initialValues: KeycloakUser;
+  serviceClass: typeof KeycloakService;
+}
+
+/** interface user action */
+export interface UserAction {
+  alias: string;
+  name: string;
+  providerId: string;
+  enabled: boolean;
+  defaultAction: boolean;
+  priority: number;
+  config: Dictionary;
+}
+
+/** default form initial values */
+export const defaultInitialValues: KeycloakUser = {
+  access: {
+    manageGroupMembership: false,
+    view: false,
+    mapRoles: false,
+    impersonate: false,
+    manage: false,
+  },
+  createdTimestamp: undefined,
+  disableableCredentialTypes: [],
+  email: '',
+  emailVerified: false,
+  enabled: true,
+  firstName: '',
+  id: '',
+  lastName: '',
+  notBefore: 0,
+  requiredActions: [],
+  totp: false,
+  username: '',
+};
+
+/** default props for editing user component */
+export const defaultProps: Partial<UserFormProps> = {
+  accessToken: 'hunter 2',
+  initialValues: defaultInitialValues,
+  serviceClass: KeycloakService,
+};
+
+/** yup validations for practitioner data object from form */
+export const userSchema = Yup.object().shape({
+  lastName: Yup.string().required('Required'),
+  firstName: Yup.string().required('Required'),
+});
+
+/** Handle required actions change */
+export const handleUserActionsChange = (
+  selected: string[],
+  setRequiredActions: Dispatch<SetStateAction<string[]>>
+): void => {
+  setRequiredActions(selected);
+};
+
+const UserForm: React.FC<UserFormProps> = (props: UserFormProps) => {
+  const { initialValues, serviceClass, accessToken } = props;
+  const [requiredActions, setRequiredActions] = React.useState<string[]>([]);
+  const [userActionOptions, setUserActionOptions] = React.useState<UserAction[]>([]);
+  const isEditMode = initialValues.id !== '';
+  const layout = {
+    labelCol: {
+      xs: { offset: 0, span: 16 },
+      sm: { offset: 2, span: 10 },
+      md: { offset: 0, span: 8 },
+      lg: { offset: 0, span: 6 },
+    },
+    wrapperCol: { xs: { span: 24 }, sm: { span: 14 }, md: { span: 12 }, lg: { span: 10 } },
+  };
+
+  const tailLayout = {
+    wrapperCol: {
+      xs: { offset: 0, span: 16 },
+      sm: { offset: 12, span: 24 },
+      md: { offset: 8, span: 16 },
+      lg: { offset: 6, span: 14 },
+    },
+  };
+  const { Option } = Select;
+
+  React.useEffect(() => {
+    const serve = new serviceClass(
+      accessToken,
+      `/authentication/required-actions/`,
+      'https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage'
+    );
+    serve
+      .list()
+      .then((response: UserAction[]) => {
+        setUserActionOptions(
+          response.filter((action: UserAction) => action.alias !== 'terms_and_conditions')
```

Comment:
Could you explain why this is here and what "terms_and_conditions" represents?

Reply:
These actions are fetched from keycloak and we are filtering it out because we dont want it as part of the user actions we set

Reply:
@kelvin-muchiri could we remove this magic string here?